tests/gstreamer_mqtt/unittest_mqtt.cc

```diff
@@ -0,0 +1,270 @@
+ */
+TEST (testMqttSink, sink_get_set_properties)
+{
+  GstHarness *h = gst_harness_new ("mqttsink");
+  gchar *sprop = NULL;
+  gboolean bprop;
+  gint iprop;
+  guint64 uprop;
+
+  ASSERT_TRUE (h != NULL);
+
+  /** test the default */
+  g_object_get (h->element, "debug", &bprop, NULL);
+  EXPECT_FALSE (bprop);
+
+  g_object_set (h->element, "debug", true, NULL);
+  g_object_get (h->element, "debug", &bprop, NULL);
+  EXPECT_TRUE (bprop);
+
+  g_object_set (h->element, "client-id", "testclientid", NULL);
+  g_object_get (h->element, "client-id", &sprop, NULL);
+  EXPECT_TRUE (g_strcmp0 (sprop, "testclientid") == 0);
+  g_free (sprop);
+
+  g_object_set (h->element, "host", "hosttest", NULL);
+  g_object_get (h->element, "host", &sprop, NULL);
+  EXPECT_TRUE (g_strcmp0 (sprop, "hosttest") == 0);
+  g_free (sprop);
+
+  g_object_set (h->element, "port", "testport", NULL);
+  g_object_get (h->element, "port", &sprop, NULL);
+  EXPECT_TRUE (g_strcmp0 (sprop, "testport") == 0);
+  g_free (sprop);
+
+  g_object_set (h->element, "pub-topic", "testtopic", NULL);
+  g_object_get (h->element, "pub-topic", &sprop, NULL);
+  EXPECT_TRUE (g_strcmp0 (sprop, "testtopic") == 0);
+  g_free (sprop);
+  
+  g_object_set (h->element, "pub-wait-timeout", 9999, NULL);
+  g_object_get (h->element, "pub-wait-timeout", &uprop, NULL);
+  EXPECT_TRUE (uprop == 9999);
+
+  g_object_set (h->element, "cleansession", false, NULL);
+  g_object_get (h->element, "cleansession", &bprop, NULL);
+  EXPECT_FALSE (bprop);
+
+  g_object_set (h->element, "keep-alive-interval", 9999, NULL);
+  g_object_get (h->element, "keep-alive-interval", &iprop, NULL);
+  EXPECT_TRUE (iprop == 9999);
+
+  g_object_set (h->element, "max-buffer-size", 1024, NULL);
+  g_object_get (h->element, "max-buffer-size", &uprop, NULL);
+  EXPECT_TRUE (uprop == 1024);
+
+  g_object_set (h->element, "num-buffers", 10, NULL);
+  g_object_get (h->element, "num-buffers", &iprop, NULL);
+  EXPECT_TRUE (iprop == 10);
+
+  g_object_set (h->element, "mqtt-qos", 1, NULL);
+  g_object_get (h->element, "mqtt-qos", &iprop, NULL);
+  EXPECT_TRUE (iprop == 1);
+
+  gst_harness_teardown (h);
+}
+
+/**
+ * @brief Test get/set properties of mqttsrc
+ */
+TEST (testMqttSrc, src_get_set_properties)
+{
+  GstHarness *h = gst_harness_new ("mqttsrc");
+  gchar *sprop = NULL;
+  gboolean bprop;
+  gint iprop;
+  gint64 lprop;
+
+  ASSERT_TRUE (h != NULL);
+
+  /** test the default */
+  g_object_get (h->element, "debug", &bprop, NULL);
+  EXPECT_FALSE (bprop);
+
+  g_object_set (h->element, "debug", true, NULL);
+  g_object_get (h->element, "debug", &bprop, NULL);
+  EXPECT_TRUE (bprop);
+
+  g_object_set (h->element, "is-live", false, NULL);
+  g_object_get (h->element, "is-live", &bprop, NULL);
+  EXPECT_FALSE (bprop);
+
+  g_object_set (h->element, "client-id", "testclientid", NULL);
+  g_object_get (h->element, "client-id", &sprop, NULL);
+  EXPECT_TRUE (g_strcmp0 (sprop, "testclientid") == 0);
```

Comment:
Please consider ```EXPECT_STREQ(str1,str2)```.

Reply:
it's applied!

Reply:
> it's applied!

Got it. I will push some commits after merging this PR.